### PR TITLE
Add AwaDB to Vector Database notebooks

### PR DESCRIPTION
**Pull request by taozhiwang**
Hi!

This pr adds a new notebook that provides an example of using Openai API and AwaDB for QA tasks.

It contains:
 - Use Openai Embedding to embed texts and save them to AwaDB
 - Do a similarity search
 - Automated prompt generation based on search results
 - Use Openai API to do question answering

### CODE REVIEW

**Review by janfilips — COMMENTED** [review]
Looks good..

**Comment by taozhiwang** [comment]
Hi @simonpfish, can you review this pr for me? Thanks for your time!

**Comment by simonpfish** [comment]
taking a look! thanks for the contribution

File: examples/vector_databases/AwaDB/Using_AwaDB_for_Question_Answering_tasks.ipynb [hunk 1]
```diff
@@ -0,0 +1,330 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "4c80dbb1",
+   "metadata": {},
+   "source": [
+    "# Using AwaDB as a Vector database for Question Answering tasks\n",
+    "\n",
+    "This notebook is an example for how you can use AwaDB as the vector database to save embedding gained from OpenAI Embedding. Then use GPT and embedding-based search to do question answering tasks\n",
```
Comment (simonpfish) [inline examples/vector_databases/AwaDB/Using_AwaDB_for_Question_Answering_tasks.ipynb:10]:
```suggestion
    "This notebook demonstrates how to utilize AwaDB as a vector database for storing embeddings obtained from OpenAI. It then illustrates how to employ GPT and embedding-based search for question answering tasks.\n",
```

File: examples/vector_databases/AwaDB/Using_AwaDB_for_Question_Answering_tasks.ipynb [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    "\n",
+    "We will provide an end-to-end workflow example to illustrate the entire process.\n",
```
Comment (simonpfish) [inline examples/vector_databases/AwaDB/Using_AwaDB_for_Question_Answering_tasks.ipynb:12]:
```suggestion
    "We outline an end-to-end workflow example to illustrate the entire process.\n",
```

**Review by simonpfish — CHANGES_REQUESTED** [review]
there are lots of misspellings and grammatical errors across the notebook. i'd suggest giving it a polish pass and then resubmitting

**Pull request closed without merging**